examples/trait-erc20/lib.rs

```diff
@@ -533,5 +533,49 @@ mod erc20 {
                 10,
             );
         }
+
+        #[ink::test]
+        fn allowance_must_not_change_on_failed_transfer() {
+            let mut erc20 = Erc20::new(100);
+            let accounts =
+                ink_env::test::default_accounts::<ink_env::DefaultEnvironment>()
+                    .expect("Cannot get accounts");
+
+            // Alice approves Bob for token transfers on her behalf.
+            let alice_balance = erc20.balance_of(accounts.alice);
+            let initial_allowance = alice_balance + 2;
+            assert_eq!(erc20.approve(accounts.bob, initial_allowance), Ok(()));
+
+            // Get contract address.
+            let callee = ink_env::account_id::<ink_env::DefaultEnvironment>()
+                .unwrap_or([0x0; 32].into());
+            // Create call.
+            let mut data =
+                ink_env::test::CallData::new(ink_env::call::Selector::new([0x00; 4])); // balance_of
+            data.push_arg(&accounts.bob);
+            // Push the new execution context to set Bob as caller.
+            assert_eq!(
+                ink_env::test::push_execution_context::<ink_env::DefaultEnvironment>(
+                    accounts.bob,
+                    callee,
+                    1000000,
+                    1000000,
+                    data
+                ),
+                ()
+            );
```

Comment:
What exactly is this `assert_eq` trying to achieve?

Reply:
That's a good question, it's used in surrounding code and ERC-721 as well. I can't make sense of it, since `push_execution_context` doesn't have a return type. So asserting for the unit type will always be true. I've created a commit to remove the assert for all these instances.

Reply:
Hard to reconstruct where it originated from due to squashed merge history. I begin to believe that squashing may not always be what we should actually be doing sometimes ...